lhotse/workflows/meeting_simulation/speaker_independent.py

```diff
@@ -218,3 +172,55 @@ def _simulate_worker(utterances: CutSet) -> MixedCut:
 
     def reverberate(self, cuts: CutSet, *rirs: RecordingSet) -> CutSet:
         return reverberate_cuts(cuts, *rirs)
+
+
+def _simulate_worker(
+    utterances: CutSet,
+    loc: float,
+    scale: float,
+    seed: int,
+) -> MixedCut:
+    # Create random number generators with the given seed.
+    npr = np.random.RandomState(seed)
+
+    # Group the cuts by speaker.
+    utts_by_speaker = defaultdict(list)
+    for utt in utterances:
+        utts_by_speaker[utt.supervisions[0].speaker].append(utt)
+
+    utterances = [CutSet.from_cuts(cuts) for cuts in utts_by_speaker.values()]
+
+    # Sample the silence durations between utterances for each speaker.
+    silence_durations = [
+        loc + npr.exponential(scale=scale, size=len(utterances[i]))
+        for i in range(len(utterances))
+    ]
+
+    # Create the meeting.
+    mixture = _create_mixture(utterances, silence_durations)
+    return mixture
+
+
+def _create_mixture(
```

Comment:
Is there a reason why this function is moved outside the class? The algorithm is specific to this particular simulator.

Reply:
Makes sense, I changed this back and used the same "trick" as in the simulator.

Reply:
Thanks. Could you do the same for `_simulate_worker` (i.e. make it a function of the specific class)? The logic for those is also tied to that algorithm.

Reply:
That would complicate the workarounds for parallelization; anyway each simulation method has a separate file, so I think it should be OK and actually consistent with other functions in lhotse that are being parallelized this way, e.g. `_X_single()` functions at the end of `lhotse/cut/set.py`.

Reply:
Okay, LGTM.